app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt

```diff
@@ -198,4 +192,8 @@ class MainFragment : Fragment() {
         _binding = null
     }
 
+    companion object {
+        fun newInstance() = MainFragment()
+    }
+
```

Comment:
https://kotlinlang.org/docs/coding-conventions.html#choose-good-names suggests companion objects be placed at the bottom of the class
I moved your companion object to the bottom of the class